src/location-tracker.ts

```diff
@@ -25,7 +25,12 @@ export class LocationTracker {
 
   getPosition(index: number): Position {
     if (index < this.lastIndex) {
-      throw new Error('Source indices must be monotonic');
+      this.lastPosition = {
+        line: 1,
+        column: 1
+      };
+
+      this.lastIndex = 0;
```

Comment:
There might be another solution to this, but the problem comes from https://github.com/posthtml/posthtml-parser/blob/41845391201256ec5e676c2d98ef292a80924b5c/src/index.ts#L138 which is easy to solve, except for when it needs to know the end of the previous line.